src/main/java/com/esri/samples/localserver/local_server_geoprocessing/LocalServerGeoprocessingController.java

```diff
@@ -0,0 +1,165 @@
+      });
+      map.getOperationalLayers().add(tiledLayer);
+
+      // listen for the status of the local server to change
+      server.addStatusChangedListener(status -> {
+        // start local geoprocessing service once local server has started
+        if (status.getNewStatus() == LocalServerStatus.STARTED) {
+          try {
+            String gpServiceURL = Paths.get(getClass().getResource("/local_server/Contour.gpk").toURI()).toString();
+            // need map server result to add contour lines to map
+            localGPService = new LocalGeoprocessingService(gpServiceURL,
+                ServiceType.ASYNCHRONOUS_SUBMIT_WITH_MAP_SERVER_RESULT);
+          } catch (URISyntaxException e) {
+            System.out.println("Failed to find gpk file. " + e.getMessage());
+          }
+
+          localGPService.addStatusChangedListener(s -> {
+            // create geoprocessing task once local geoprocessing service is started
+            if (s.getNewStatus() == LocalServerStatus.STARTED) {
+              // add `/Contour` to use contour geoprocessing tool 
+              gpTask = new GeoprocessingTask(localGPService.getUrl() + "/Contour");
+              btnClear.disableProperty().bind(btnGenerate.disabledProperty().not());
+              btnGenerate.setDisable(false);
+            }
+          });
+          localGPService.startAsync();
+        }
+      });
+      server.startAsync();
+
+    } catch (Exception e) {
+      // on any exception, print the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   *  Creates a Map Image Layer that displays contour lines on the map using the interval the that is set. 
+   *   
+   */
+  @FXML
+  protected void handleGenerateContours() {
+
+    // tracking progress of creating contour map 
+    progressBar.setVisible(true);
+    // create parameter using interval set
+    double interval = Double.parseDouble(txtInterval.getText());
+    GeoprocessingParameters gpParameters = new GeoprocessingParameters(
+        GeoprocessingParameters.ExecutionType.ASYNCHRONOUS_SUBMIT);
+    final Map<String, GeoprocessingParameter> inputs = gpParameters.getInputs();
+    inputs.put("Interval", new GeoprocessingDouble(interval));
+
+    // adds contour lines to map
+    GeoprocessingJob gpJob = gpTask.createJob(gpParameters);
+    gpJob.addJobDoneListener(() -> {
+      if (gpJob.getStatus() == Job.Status.SUCCEEDED) {
+        // creating map image url from local groprocessing service url
+        String serviceUrl = localGPService.getUrl();
+        String mapServerUrl = serviceUrl.replace("GPServer", "MapServer/jobs/" + gpJob.getServerJobId());
+        ArcGISMapImageLayer mapImageLayer = new ArcGISMapImageLayer(mapServerUrl);
+        mapImageLayer.loadAsync();
+        mapView.getMap().getOperationalLayers().add(mapImageLayer);
+        btnGenerate.setDisable(true);
+      } else {
+        System.out.println("Error: " + gpJob.getError().getAdditionalMessage());

```

Comment:
This will not be visible in sample viewer. Should probably show the message in a dialog like the editing samples.